pandas/tests/reshape/merge/test_join.py

```diff
@@ -956,3 +956,35 @@ def test_join_empty(left_empty, how, exp):
             expected = expected.rename_axis("A")
 
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("how", ["inner", "outer", "left", "right"])
+def test_join_multiindex_categorical_output_index_dtype(how):
+    # GH#50906
+    df1 = DataFrame(
+        {
+            "idx1": Categorical(["a", "a", "a"]),
+            "idx2": Categorical(["a", "a", "b"]),
+            "data": [1, 2, 3],
+        }
+    ).set_index(["idx1", "idx2"])
```

Comment:
thanks for working on this 

why not use the example from #50906? for regression tests, if the original example is small and self-contained, it'd probably be safer to just use that directly
the rest looks good though 👍